src/core/bau_inputs.jl

```diff
@@ -164,15 +175,134 @@ function BAUInputs(p::REoptInputs)
+            bau_emissions_lb_SO2_per_year, bau_emissions_lb_PM25_per_year, bau_grid_emissions_lb_CO2_per_year, 
+            bau_grid_emissions_lb_NOx_per_year, bau_grid_emissions_lb_SO2_per_year, bau_grid_emissions_lb_PM25_per_year)
+    
+    s.site.bau_emissions_lb_CO2_per_year = bau_emissions_lb_CO2_per_year
+    s.site.bau_emissions_lb_NOx_per_year = bau_emissions_lb_NOx_per_year
+    s.site.bau_emissions_lb_SO2_per_year = bau_emissions_lb_SO2_per_year
+    s.site.bau_emissions_lb_PM25_per_year = bau_emissions_lb_PM25_per_year
+
+    s.site.bau_grid_emissions_lb_CO2_per_year = bau_grid_emissions_lb_CO2_per_year
+    s.site.bau_grid_emissions_lb_NOx_per_year = bau_grid_emissions_lb_NOx_per_year
+    s.site.bau_grid_emissions_lb_SO2_per_year = bau_grid_emissions_lb_SO2_per_year
+    s.site.bau_grid_emissions_lb_PM25_per_year = bau_grid_emissions_lb_PM25_per_year
+end
+
+"""
+    setup_bau_emissions_inputs(p::REoptInputs, s_bau::BAUScenario, generator_fuel_use_gal::Float64)
+
+Pre-processing of the BAU emissions to use in determining emissions reductions in the optimal case
+Include BAU grid emissions, existing backup generator emissions, boiler emissions
+Update the `bau_(grid_)emissions_` values in s.site and s_bau.site
+
+Note if existing generation does not sustain a simulated outage, the BAU load and therefore emissions
+are 0 during unsurvived outage hours
+"""
+function setup_bau_emissions_inputs(p::REoptInputs, s_bau::BAUScenario, generator_fuel_use_gal::Float64)
+    
+    bau_emissions_lb_CO2_per_year = 0
+    bau_emissions_lb_NOx_per_year = 0
+    bau_emissions_lb_SO2_per_year = 0
+    bau_emissions_lb_PM25_per_year = 0
+
+    ## Grid emissions
+
+    # This function is called after ajust_load_profile() (makes load_kw native if not already) 
+    # so need to calculate net version of loads_kw by removing existing PV to get load served by the grid.
+    # TODO: Should load_profile have loads_kw and loads_kw_net fields, instead of updating loads_kw? Would be easier 
+    # to keep track of what version you're using, don't have to make sure whether it's before or after the adjustment.
+    # Must account for levelization factor to align with how PV is modeled in REopt:
+    # Because we only model one year, we multiply the "year 1" PV production by a levelization_factor
+    # that accounts for the PV capacity degradation over the analysis_years. In other words, by
+    # multiplying the pv production_factor by the levelization_factor we are modeling the average pv production.
+    bau_grid_to_load = copy(s_bau.electric_load.loads_kw)
+    bau_grid_to_load_critical = copy(s_bau.electric_load.critical_loads_kw)
+    for pv in p.s.pvs if pv.existing_kw > 0
+        bau_grid_to_load .-= p.levelization_factor[pv.name] * pv.existing_kw * p.production_factor[pv.name, :].data
+    end end
+    for pv in p.s.pvs if pv.existing_kw > 0
```

Comment:
Can these two for-if statements be combined into one?

Reply:
yes